lld/MachO/Driver.cpp

```diff
@@ -1295,6 +1394,28 @@ static void createFiles(const InputArgList &args) {
       break;
     }
   }
+
+  if (config->readThreads) {
+    multiThreadedPageIn(deferredFiles);
+
+    DeferredFiles archiveContents;
+    std::vector<ArchiveFile *> archives;
+    for (auto &file : deferredFiles) {
+      auto inputFile = processFile(file.buffer, &archiveContents, file.path,
+                                   LoadType::CommandLine, file.isLazy);
+      if (ArchiveFile *archive = dyn_cast<ArchiveFile>(inputFile))
+        archives.push_back(archive);
+    }
+
+    if (!archiveContents.empty()) {
+      multiThreadedPageIn(archiveContents);
+      for (auto *archive : archives)
+        archive->addLazySymbols();
+    }
+
+    // reap threads
+    // multiThreadedPageIn();
```

Comment:
Not sure if you still want to keep the `join` of the threads here. In one of your comments you seem to imply that you want, but the last commit seems to imply that it doesn't change things.

If you still want to keep it, my opinion is still that it is complicated to understand what `multiThreadedPageIn()` without parameters or with an empty parameter is actually doing and how it is "reaping" threads until you read the implementation of `multiThreadedPageIn` carefully, when a more clear function called "join" will make it very obvious.